uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -63,236 +71,179 @@ pub struct KotlinWrapper<'a> {
 
-    /// Get the Kotlin syntax for representing a given api-level `Type`.
-    pub fn type_kt(type_: &Type) -> Result<String, askama::Error> {
-        Ok(match type_ {
-            // These native Kotlin types map nicely to the FFI without conversion.
-            Type::UInt8 => "UByte".to_string(),
-            Type::UInt16 => "UShort".to_string(),
-            Type::UInt32 => "UInt".to_string(),
-            Type::UInt64 => "ULong".to_string(),
-            Type::Int8 => "Byte".to_string(),
-            Type::Int16 => "Short".to_string(),
-            Type::Int32 => "Int".to_string(),
-            Type::Int64 => "Long".to_string(),
-            Type::Float32 => "Float".to_string(),
-            Type::Float64 => "Double".to_string(),
-            // These types need conversion, and special handling for lifting/lowering.
-            Type::Boolean => "Boolean".to_string(),
-            Type::String => "String".to_string(),
-            Type::Timestamp => "java.time.Instant".to_string(),
-            Type::Duration => "java.time.Duration".to_string(),
-            Type::Enum(name)
-            | Type::Record(name)
-            | Type::Object(name)
-            | Type::Error(name)
-            | Type::CallbackInterface(name) => class_name_kt(name)?,
-            Type::Optional(t) => format!("{}?", type_kt(t)?),
-            Type::Sequence(t) => format!("List<{}>", type_kt(t)?),
-            Type::Map(t) => format!("Map<String, {}>", type_kt(t)?),
-        })
-    }
-
-    /// Get the Kotlin syntax for representing a given low-level `FFIType`.
-    pub fn type_ffi(type_: &FFIType) -> Result<String, askama::Error> {
-        Ok(match type_ {
-            // Note that unsigned integers in Kotlin are currently experimental, but java.nio.ByteBuffer does not
-            // support them yet. Thus, we use the signed variants to represent both signed and unsigned
-            // types from the component API.
-            FFIType::Int8 | FFIType::UInt8 => "Byte".to_string(),
-            FFIType::Int16 | FFIType::UInt16 => "Short".to_string(),
-            FFIType::Int32 | FFIType::UInt32 => "Int".to_string(),
-            FFIType::Int64 | FFIType::UInt64 => "Long".to_string(),
-            FFIType::Float32 => "Float".to_string(),
-            FFIType::Float64 => "Double".to_string(),
-            FFIType::RustArcPtr => "Pointer".to_string(),
-            FFIType::RustBuffer => "RustBuffer.ByValue".to_string(),
-            FFIType::ForeignBytes => "ForeignBytes.ByValue".to_string(),
-            FFIType::ForeignCallback => "ForeignCallback".to_string(),
-        })
-    }
+impl KotlinLanguageOracle {
+    fn create_code_type(&self, type_: TypeIdentifier) -> Box<dyn CodeType> {
+        // I really want access to the ComponentInterface here so I can look up the interface::{Enum, Record, Error, Object, etc}
+        // However, there's some violence and gore I need to do to (temporarily) make the oracle usable from filters.
```

Comment:
Lol, I know what you mean - it would be nice to be able to have backward references from the leaf nodes here (`Enum`, `Record`, etc) to the containing `ComponentInterface`.

Reply:
Also, if we were able to do this in a pre-processing step, it might avoid some of this awkward passing around of global context :-)

Reply:
I can see having a pre-processing step would be nice.

I've recently come a-cropper with `ci.type_contains_object_references` and `member.contains_unsigned_types(ci)`. It would be nice if I didn't need to calculate this all the time.

Reply:
This is one of those cases where having parent pointers in the member structs would be valuable as well.

Reply:
I'm not convinced that parent pointers would be worth the hassle. 

I'm able to make `MemberDeclaration`s query the `ci` at their construction. 

What I _am_ finding problematic is "Does this `Type::` need an `@ExperimentalUnsignedTypes` annotation, i.e. going from `Type` to  CI member.